### DISCUSSION TITLE
Ellar Serializer or ModelSchema (from ninja_schema)

### DISCUSSION THREAD

**Discussion by paolodina** [python-ellar/ellar#191] [Q&A] [answered]
I'm starting writing  schemas for my app, I wonder if I can use `ModelSchema` from [ninja_schema](https://github.com/eadwinCode/ninja-schema) instead of Ellar's `Serializer` as shown in the docs. Using `ModelSchema` classes I could generate schemas from Django models, which is very handy, but I'm not sure I can do that using `Serializer`s as well.

PS. I forgot to say that (obviously) I'm using Ellar with Django via https://github.com/python-ellar/ellar-django-module

**Comment by eadwinCode** [comment]
Hi @paolodina Thanks for trying out this project.

The `Serializer` was not designed to server the same purpose as ninja_schema.  So you cant use `Serializer` in place of `ModelSchema`.

On the other side, ninja_schema creates a pydantic model at the end of its setup. So, I think you can use it as an input/output schema for your controller. Ellar will always convert to serializer model if its a pydantic type.

**Comment by paolodina** [comment]
> Hi @paolodina Thanks for trying out this project.

Which is great. (*)

> The `Serializer` was not designed to server the same purpose as ninja_schema. So you cant use `Serializer` in place of `ModelSchema`.
> 
> On the other side, ninja_schema creates a pydantic model at the end of its setup. So, I think you can use it as an input/output schema for your controller. Ellar will always convert to serializer model if its a pydantic type.

That's exactly how I'm utilizing the schemas now, and the controllers can use them without any issues. My confusion, which led me to try ninja_schema, stemmed from seeing some code examples that employed a DTO (btw, `EventDTO`), which I'd like to generate from a Django model. I had difficulty understanding the need for DTOs in this context, and how to effectively use them, as instinctively, Django models seemed to be sufficient for my needs.

---

(*) I started the last project using django-ninja-extras, but found Ellar and restarted with it. But, what's your plans for them? What should I use?

**Reply by eadwinCode** [reply, marked as answer]
>That's exactly how I'm utilizing the schemas now, and the controllers can use them without any issues. My confusion, which led me to try ninja_schema, stemmed from seeing some code examples that employed a DTO (btw, EventDTO), which I'd like to generate from a Django model. I had difficulty understanding the need for DTOs in this context, and how to effectively use them, as instinctively, Django models seemed to be sufficient for my needs.

Oh okay thats nice. So ellar-with-django project was designed to use a repository pattern. The repository interface requires a DTO to define data transfer schema. So `EventDTO` is there to serve the purpose.

>(*) I started the last project using django-ninja-extras, but found Ellar and restarted with it. But, what's your plans for them? What should I use?

Both projects will be maintained. 

`django-ninja-extra` was design to add more flavour to `django-ninja` and for DRF users to easily port their code base to `django-ninja`. Also a lot of DRF libraries needs to be converted using `django-ninja-extra` as base because the library already provides similar pattern as DRF. 
I have only converted [`password-reset`](https://github.com/eadwinCode/django-ninja-passwordreset) package. So with that, anyone can use that as a reference to convert other libraries

In terms of which to use, use ellar, although Django has lot more batteries for quick start. The project will catch up with that soon enough. But if you want quick start up and all that django has to offer, django-ninja-extra is good. Ellar at this stage cant match with django

if there are other questions about Ellar or suggestions, please do share. Thanks
